[GUI] Settings information, fix missing initial masternodes count value.

furszy opened the pull request:
The count isn't updated on every widget show, only via the event (which only happen every 40 minutes as Masternodes updates aren't regular at all).
This PR fixes it updating the masternode count every time that the screen is visible, then the event is in charge of update it if the user stays there long enough.

random-zebra reviewed (commented):
Won't this just set the masternode count to an empty string?
`cachedMasternodeCountString` is empty till `updateMnTimer` is called (and is able to acquire `cs_main` lock).

Wouldn't it be better to call `updateMnTimer` immediately from within `startMasternodesTimer` ? (though it would probably fail to lock cs_main anyway, and the widget would still display an empty string for some time).

furszy commented:
Updated the PR, feedback tackled.
 Moved the forced count update to a worker thread due the internal mnmanger mutex requirement. Plus removed the unneeded `cs_main` lock from the GUI as mnmanger is already guarding the internal map with its own mutex.

random-zebra reviewed (approved):
Nice. ACK 164d6aa7ce3946e34f00d8993da81a3e6cd81953

Fuzzbawls reviewed (approved):
ACK 164d6aa7ce3946e34f00d8993da81a3e6cd81953
